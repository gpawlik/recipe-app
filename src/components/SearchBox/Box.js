import styled from 'styled-components';

const Box = styled.div`
  margin: 0 auto;
  max-width: 800px;
  display: flex;
`;

export default Box;
