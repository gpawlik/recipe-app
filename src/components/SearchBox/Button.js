import styled from 'styled-components';

import { colors } from 'variables';

const Button = styled.button`
  color: ${colors.MIDNIGHT_BLACK};
  background-color: ${colors.WHITE};
  font-size: 16px;
  font-weight: 600;
  padding: 0 1em;
  margin-left: 5px;
  line-height: 30px;
  cursor: pointer;
  border-radius: 3px;
  border: 1px solid ${colors.DOVE_GRAY};
  box-shadow: 0px 0px -1px 4px rgba(0, 0, 0, 0.2);

  &:hover {
    background-color: ${colors.DOVE_GRAY};
  }
`;

export default Button;
