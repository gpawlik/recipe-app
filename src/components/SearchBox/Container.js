import styled from 'styled-components';

import { colors } from 'variables';

const Container = styled.div`
  background-color: ${colors.GREEN_SEA};
  padding: 10px 15px;
`;

export default Container;
