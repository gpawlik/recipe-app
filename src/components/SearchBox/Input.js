import styled from 'styled-components';

import { colors } from 'variables';

const Input = styled.input`
  border: 1px solid ${colors.DOVE_GRAY};
  border-radius: 3px;
  font-size: 16px;
  line-height: 30px;
  padding: 0 10px;
  flex-grow: 1;
`;

export default Input;
