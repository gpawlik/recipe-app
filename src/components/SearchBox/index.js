import React from 'react';
import PropTypes from 'prop-types';

import Container from './Container';
import Box from './Box';
import Input from './Input';
import Button from './Button';

const Searchbox = ({ value, onChange, onSearch, isDisabled }) => {
  return (
    <Container>
      <Box>
        <Input type="search" value={value} onChange={onChange} />
        <Button onClick={onSearch} disabled={isDisabled}>Search</Button>
      </Box>
    </Container>
  );
};

Searchbox.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool
};

export default Searchbox;
