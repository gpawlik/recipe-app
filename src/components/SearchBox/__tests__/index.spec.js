import React from 'react';
import { create } from 'react-test-renderer';

import SearchBox from '../';

const initialProps = {
  onChange: () => {},
  onSearch: () => {}
};

const mockComponent = props => {
  const finalProps = {
    ...initialProps,
    ...props
  };

  return (
    <SearchBox {...finalProps} />
  );
};

describe('SearchBox', () => {
  it('should render with the default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with custom input value', () => {
    const props = {
      value: 'foo'
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with disabled button', () => {
    const props = {
      isDisabled: true
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
