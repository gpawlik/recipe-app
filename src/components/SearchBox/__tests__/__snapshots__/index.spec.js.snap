// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render with custom input value 1`] = `
<div
  className="sc-bdVaJa aSJuf"
>
  <div
    className="sc-bwzfXH kQxmPG"
  >
    <input
      className="sc-htpNat gctxNE"
      onChange={[Function]}
      type="search"
      value="foo"
    />
    <button
      className="sc-bxivhb ezHTTB"
      disabled={undefined}
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`SearchBox should render with disabled button 1`] = `
<div
  className="sc-bdVaJa aSJuf"
>
  <div
    className="sc-bwzfXH kQxmPG"
  >
    <input
      className="sc-htpNat gctxNE"
      onChange={[Function]}
      type="search"
      value={undefined}
    />
    <button
      className="sc-bxivhb ezHTTB"
      disabled={true}
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`SearchBox should render with the default props 1`] = `
<div
  className="sc-bdVaJa aSJuf"
>
  <div
    className="sc-bwzfXH kQxmPG"
  >
    <input
      className="sc-htpNat gctxNE"
      onChange={[Function]}
      type="search"
      value={undefined}
    />
    <button
      className="sc-bxivhb ezHTTB"
      disabled={undefined}
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
